src/OrchardCore.Modules/OrchardCore.Roles/ViewModels/PermissionGroupKey.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+
+namespace OrchardCore.Roles.ViewModels;
+
+public class PermissionGroupKey
+{
+    private readonly string _key;
+
+    public string Title { get; set; }
+
+    public string Source { get; set; }
+
+    public PermissionGroupKey(string key)
+    {
+        if (String.IsNullOrWhiteSpace(key))
+        {
+            throw new ArgumentException(nameof(key));
+        }
+
+        _key = key;
+    }
+
+    public PermissionGroupKey(string key, string title)
```

Comment:
public PermissionGroupKey(string key, string title) : this(key) => Title = title;